handlers/handlers.go

```diff
@@ -150,62 +202,203 @@ func (d *CatalystAPIHandlersCollection) processUploadVOD(streamName, sourceURL,
-		if err != nil {
-			errors.WriteHTTPInternalServerError(w, "Cannot read payload", err)
-			return
+		// Multiple pushes from RENDITION_PREFIX are in progress.
+		return
+	}
+	tracks := make(LiveTrackListTriggerJson)
+	if err = json.Unmarshal([]byte(encodedTracks), &tracks); err != nil {
+		errorOutInternal("LiveTrackListTriggerJson json decode error", err)
+		return
+	}
+	// Start push per each video track
+	info, err := d.StreamCache.Transcoding.Get(streamName)
+	if err != nil {
+		errorOutInternal("LIVE_TRACK_LIST unknown push source", err)
+		return
+	}
+	for i := range tracks { // i is generated name, not important, all info is contained in element
+		if tracks[i].Type != "video" {
+			// Only produce an rendition per each video track, selecting best audio track
+			continue
 		}
-		lines := strings.Split(strings.TrimSuffix(string(payload), "\n"), "\n")
-		if len(lines) < 2 {
-			errors.WriteHTTPBadRequest(w, "Bad request payload", fmt.Errorf("unknown payload '%s'", string(payload)))
-			return
+		destination := fmt.Sprintf("%s/%s__%dx%d.ts?video=%d&audio=maxbps", info.UploadDir, streamName, tracks[i].Width, tracks[i].Height, tracks[i].Index) //.Id)
+		fmt.Printf("> Starting push to %s\n", destination)
+		if err := d.MistClient.PushStart(streamName, destination); err != nil {
+			fmt.Printf("> ERROR push to %s %v\n", destination, err)
+		} else {
+			d.StreamCache.Transcoding.AddDestination(streamName, destination)
 		}
+	}
+	fmt.Printf("> Trigger_LIVE_TRACK_LIST %v\n", lines)
+}
+
+// This trigger is run whenever an outgoing push stops, for any reason.
+// This trigger is stream-specific and non-blocking. containing data as such:
+//   push ID (integer)
+//   stream name (string)
+//   target URI, before variables/triggers affected it (string)
+//   target URI, afterwards, as actually used (string)
+//   last 10 log messages (JSON array string)
+//   most recent push status (JSON object string)
+func (d *MistCallbackHandlersCollection) Trigger_PUSH_END(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	streamName := "unknown"
+	errorOutInternal := func(where string, err error) {
+		txt := fmt.Sprintf("%s name=%s", where, streamName)
+		// Mist does not respond or handle returned error codes. We do this for correctness.
+		errors.WriteHTTPInternalServerError(w, txt, err)
+		fmt.Printf("<ERROR LIVE_TRACK_LIST> %s %v\n", txt, err)
+	}
 
-		// stream name is the second line in the Mist Trigger payload
-		s := lines[1]
-		// when uploading is done, remove trigger and stream from Mist
-		errT := d.MistClient.DeleteTrigger(s, "PUSH_END")
-		errS := d.MistClient.DeleteStream(s)
-		if errT != nil {
-			errors.WriteHTTPInternalServerError(w, fmt.Sprintf("Cannot remove PUSH_END trigger for stream '%s'", s), errT)
+	payload, err := io.ReadAll(req.Body)
+	if err != nil {
+		errorOutInternal("Cannot read payload", err)
+		return
+	}
+	lines := strings.Split(strings.TrimSuffix(string(payload), "\n"), "\n")
+	if len(lines) < 2 {
```

Comment:
I think this condition should be different, because you read more lines than `2`.